New topic - show or hide password checkbox implementation

ajithkumar-maragathavel opened the pull request:
## Purpose of this pull request

This pull request (PR) adds a new topic that describes the implementation of the show or the hide password checkbox functionality.

## Affected DevDocs pages

<!-- REQUIRED List the affected pages on devdocs.magento.com (URLs). Not needed for large numbers of files. -->

-  New Topic

## Links to Magento source code

<!--  OPTIONAL - REMOVE THIS SECTION IF NOT USED. If this pull request references a file in a Magento codebase repository, add it here. -->

-  https://github.com/magento/magento2/pull/31482

Fixes #8808 

whatsnew
Added the [Show or Hide Password Checkbox](https://devdocs.magento.com/guides/v2.4/ui_comp_guide/howto/show_or_hide_password_checkbox.html) topic.

BarnyShergold commented on src/guides/v2.3/ui_comp_guide/howto/show_or_hide_password_checkbox.md:
```diff
@@ -0,0 +1,103 @@
+<div class="block-content" aria-labelledby="block-company-login-heading">
+    <form class="form form-company-login" action="<?= $block->escapeUrl($block->getPostActionUrl()) ?>" method="post" id="company-login-form">
+        <?= $block->getBlockHtml('formkey') ?>
+        <fieldset class="fieldset login" data-hasrequired="<?= $block->escapeHtml(__('* Required Fields')) ?>">
```
Block Escaper is deprecated.
```suggestion
        <fieldset class="fieldset login" data-hasrequired="<?= $escaper->escapeHtml(__('* Required Fields')) ?>">
```

BarnyShergold commented on src/guides/v2.3/ui_comp_guide/howto/show_or_hide_password_checkbox.md:
````diff
@@ -0,0 +1,103 @@
+
+```phtml
+<div class="block-content" aria-labelledby="block-company-login-heading">
+    <form class="form form-company-login" action="<?= $block->escapeUrl($block->getPostActionUrl()) ?>" method="post" id="company-login-form">
````
```suggestion
    <form class="form form-company-login" action="<?= $escaper->escapeUrl($block->getPostActionUrl()) ?>" method="post" id="company-login-form">
```

BarnyShergold commented on src/guides/v2.3/ui_comp_guide/howto/show_or_hide_password_checkbox.md:
```diff
@@ -0,0 +1,103 @@
+        <?= $block->getBlockHtml('formkey') ?>
+        <fieldset class="fieldset login" data-hasrequired="<?= $block->escapeHtml(__('* Required Fields')) ?>">
+            <div class="field email required">
+                <label class="label" for="email"><span><?= $block->escapeHtml(__('Company Email')) ?></span></label>
```
```suggestion
                <label class="label" for="email"><span><?= $escaper->escapeHtml(__('Company Email')) ?></span></label>
```

BarnyShergold commented on src/guides/v2.3/ui_comp_guide/howto/show_or_hide_password_checkbox.md:
```diff
@@ -0,0 +1,103 @@
+                </div>
+            </div>
+            <div class="field password required">
+                <label for="pass" class="label"><span><?= $block->escapeHtml(__('Password')) ?></span></label>
```
```suggestion
                <label for="pass" class="label"><span><?= $escaper->escapeHtml(__('Password')) ?></span></label>
```

BarnyShergold commented on src/guides/v2.3/ui_comp_guide/howto/show_or_hide_password_checkbox.md:
```diff
@@ -0,0 +1,103 @@
+            </div>
+            <div class="actions-toolbar">
+                <div class="primary">
+                    <button type="submit" class="action login primary" id="company-submit"><span><?= $block->escapeHtml(__('Company Sign In')) ?></span></button>
```
```suggestion
                    <button type="submit" class="action login primary" id="company-submit"><span><?= $escaper->escapeHtml(__('Company Sign In')) ?></span></button>
```

BarnyShergold reviewed (changes requested):
Great new topic explanation!

ajithkumar-maragathavel commented:
@BarnyShergold Thanks for the review. I have updated the PR based on your comments. Please have a look at it.

BarnyShergold reviewed (approved):
Nice!

dobooth commented:
Nice work, @ajithkumar-maragathavel. Rather than symlink the image, you can put it in `src/common/images` and call it from there in both versions.

ajithkumar-maragathavel commented:
> Nice work, @ajithkumar-maragathavel. Rather than symlink the image, you can put it in `src/common/images` and call it from there in both versions.

@dobooth  Thanks for the suggestion. I have updated the PR by moving the image to the common directory and removed symlinks as well. Please review it.

dobooth commented:
@magento import pr to magento-devdocs/devdocs

magento-engcom-team commented:
@dobooth the pull request successfully imported.

ajithkumar-maragathavel commented:
@dobooth I have updated the new topic only for v2.4. Please review it.

dobooth commented:
@magento import code to magento-devdocs/devdocs

atwixfirster commented:
Hi,  ajithkumar-maragathavel !

Thank you for your work here. Sorry for being a bit late on the party but as a reader I would like to know what is a `Magento_Customer/js/show-password` component and where is I can find it in Magento codebase?

Thank you!

ajithkumar-maragathavel commented:
> Hi, @ajithkumar-maragathavel !
> 
> Thank you for your work here. Sorry for being a bit late on the party but as a reader I would like to know what is a `Magento_Customer/js/show-password` component and where is I can find it in Magento codebase?
> 
> Thank you!

@atwixfirster  Thanks for the suggestion. I will update the document in a separate PR explaining the `Magento_Customer/js/show-password` component and I will also add the Magento codebase link.

atwixfirster commented:
> > Hi, @ajithkumar-maragathavel !
> > Thank you for your work here. Sorry for being a bit late on the party but as a reader I would like to know what is a `Magento_Customer/js/show-password` component and where is I can find it in Magento codebase?
> > Thank you!
> 
> @atwixfirster Thanks for the suggestion. I will update the document in a separate PR explaining the `Magento_Customer/js/show-password` component and I will also add the Magento codebase link.

Cool. Thank you, @ajithkumar-maragathavel
